### PR TITLE
Issue 35901: explode(): Argument #2 () must be of type string, null g…

### LINKED ISSUE #35901: TypeError: explode(): Argument #2 ($string) must be of type string, null given in /var/www/php/folder/vendor/magento/module-config/ViewModel/CountryFilter.php:46
### Preconditions and environment

- Magento version - 2.4.5 CE
- Visit Stores -> Configuration and see error 
- Error reporting turned on
- Page shows an error
- PHP 8.1.9

### Steps to reproduce

1. Login into Admin Site
2. Go to Stores->Configuration

### Expected result

Configuration page visible

### Actual result

Error is thrown:

TypeError: explode(): Argument #2 ($string) must be of type string, null given in /var/www/php/folder/vendor/magento/module-config/ViewModel/CountryFilter.php:46
Stack trace:
#0 /var/www/php/folder/vendor/magento/module-config/ViewModel/CountryFilter.php(46): explode(',', NULL)
#1 /var/www/php/folder/vendor/magento/module-config/view/adminhtml/templates/system/config/js.phtml(81): Magento\Config\ViewModel\CountryFilter->getEuCountryList()
#2 /var/www/php/folder/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('/var/www/php/mo...')
#3 /var/www/php/folder/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Backend\Block\Template), '/var/www/php/mo...', Array)
#4 /var/www/php/folder/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent('render', Array)
#5 /var/www/php/folder/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\TemplateEngine\Php\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Backend\Block\Template), '/var/www/php/mo...', Array)
#6 /var/www/php/folder/generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php(23): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callPlugins('render', Array, Array)
#7 /var/www/php/folder/vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php\Interceptor->render(Object(Magento\Backend\Block\Template), '/var/www/php/mo...', Array)
#8 /var/www/php/folder/vendor/magento/framework/View/Element/Template.php(293): Magento\Framework\View\Element\Template->fetchView('/var/www/php/mo...')
#9 /var/www/php/folder/vendor/magento/module-backend/Block/Template.php(141): Magento\Framework\View\Element\Template->_toHtml()
#10 /var/www/php/folder/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Backend\Block\Template->_toHtml()
#11 /var/www/php/folder/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#12 /var/www/php/folder/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#13 /var/www/php/folder/vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#14 /var/www/php/folder/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock('js.system_confi...')
#15 /var/www/php/folder/generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout->renderNonCachedElement('js.system_confi...')
#16 /var/www/php/folder/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('js.system_confi...')
#17 /var/www/php/folder/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement('js.system_confi...', false)
#18 /var/www/php/folder/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('js', false)
#19 /var/www/php/folder/generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout->renderNonCachedElement('js')
#20 /var/www/php/folder/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('js')
#21 /var/www/php/folder/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement('js', false)
#22 /var/www/php/folder/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('page.wrapper', false)
#23 /var/www/php/folder/generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#24 /var/www/php/folder/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.wrapper')
#25 /var/www/php/folder/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement('page.wrapper', false)
#26 /var/www/php/folder/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('backend.page', false)
#27 /var/www/php/folder/generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout->renderNonCachedElement('backend.page')
#28 /var/www/php/folder/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('backend.page')
#29 /var/www/php/folder/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement('backend.page', false)
#30 /var/www/php/folder/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('root', false)
#31 /var/www/php/folder/generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout->renderNonCachedElement('root')
#32 /var/www/php/folder/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('root')
#33 /var/www/php/folder/vendor/magento/framework/View/Layout.php(975): Magento\Framework\View\Layout->renderElement('root')
#34 /var/www/php/folder/generated/code/Magento/Framework/View/Layout/Interceptor.php(50): Magento\Framework\View\Layout->getOutput()
#35 /var/www/php/folder/vendor/magento/framework/View/Result/Page.php(260): Magento\Framework\View\Layout\Interceptor->getOutput()
#36 /var/www/php/folder/vendor/magento/framework/View/Result/Layout.php(171): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#37 /var/www/php/folder/generated/code/Magento/Backend/Model/View/Result/Page/Interceptor.php(32): Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#38 /var/www/php/folder/vendor/magento/framework/App/Http.php(120): Magento\Backend\Model\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#39 /var/www/php/folder/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#40 /var/www/php/folder/pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#41 {main}

### Additional information

_No response_

### Release note

_No response_

### Triage and priority

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by rogerdz**
…iven

### Description (*)

TypeError: explode(): Argument #2 ($string) must be of type string, null given in /var/www/php/folder/vendor/magento/module-config/ViewModel/CountryFilter.php:46

### Fixed Issues (if relevant)

1. Fixes magento/magento2#35901

### Manual testing scenarios (*)

1. Login into Admin Site
2. Go to Stores->Configuration->General->General
3. Uncheck "Use system value" European Union Countries, remove all select
4. Save config

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogerdz** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/App/Config.php [hunk 1]
```diff
@@ -132,6 +134,6 @@ public function get($configType, $path = '', $default = null)
             $result = $this->types[$configType]->get($path);
         }
 
-        return $result !== null ? $result : $default;
+        return $result !== null ? $result : ($default !== null ? $default : '');
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/App/Config.php]: I think we can simplify it 
```suggestion
        return $result ?? $default ?? '';
```

Here is an example:
https://3v4l.org/SY8Ys

Cloud you, please also cover this change with a unit test?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/App/Config.php (hunk 1 above)
Comment (rogerdz) [reply lib/internal/Magento/Framework/App/Config.php]: @ihor-sviziev I added unit test

**Comment by rogerdz** [comment]
@magento run all tests

**Comment by rogerdz** [comment]
@magento run all tests

**Comment by rogerdz** [comment]
This change seems to affect many places, so I will close the PR and create another PR to fix the error more easily

File: lib/internal/Magento/Framework/App/Config.php [hunk 2]
```diff
@@ -1,25 +1,25 @@
 <?php
 /**
- * Application configuration object. Used to access configuration when application is initialized and installed.
- *
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/App/Config.php]: I think declaring strict types can broke tests. Let's try to revert it

Comment (rogerdz) [reply lib/internal/Magento/Framework/App/Config.php]: Thank @ihor-sviziev, I reverted it

**Comment by rogerdz** [comment]
@magento run all tests

**Comment by rogerdz** [comment]
Change return value has a lot of impact so I will close this PR and create a new PR that is easier to fix bug

**Pull request closed without merging**